src/tools/rbd/action/Snap.cc

```diff
@@ -35,18 +35,23 @@ int do_list_snaps(librbd::Image& image, Formatter *f)
     t.define_column("SNAPID", TextTable::RIGHT, TextTable::RIGHT);
     t.define_column("NAME", TextTable::LEFT, TextTable::LEFT);
     t.define_column("SIZE", TextTable::RIGHT, TextTable::RIGHT);
+    t.define_column("TIMESTAMP", TextTable::RIGHT, TextTable::RIGHT);
   }
 
   for (std::vector<librbd::snap_info_t>::iterator s = snaps.begin();
        s != snaps.end(); ++s) {
+    std::string timestamp;
+    image.snap_get_timestamp(s->id, &timestamp);
+
     if (f) {
       f->open_object_section("snapshot");
       f->dump_unsigned("id", s->id);
       f->dump_string("name", s->name);
       f->dump_unsigned("size", s->size);
+      f->dump_string("timestamp", timestamp);
```

Comment:
Make sure you aren't going to dump "0" time for old snapshots -- just output nothing when the timestamp is zeroed.